pynautobot/core/endpoint.py

```diff
@@ -81,7 +81,7 @@ def _lookup_ret_obj(self, name, model):
             ret = Record
         return ret
 
-    def all(self, api_version=None, limit=None, offset=None):
+    def all(self, *args, **kwargs):
```

Comment:
Hmm, I wonder if we should instead try and preserve the current signature for better backwards compatibility.

For example, if you call `nb.dcim.devices.all("1.2", 100, 0)` in the current version it is equivalent to `nb.dcim.devices.all(api_version="1.2", limit=100, offset=0)`. I don't think we need `*args` for this call so maybe just adding `**kwargs` is all we need here:
```suggestion
    def all(self, api_version=None, limit=None, offset=None, **kwargs):
```

Thoughts?

We would also need to change the call to filter:
```python
return self.filter(api_version=api_version, limit=limit, offset=offset, **kwargs)
```

Reply:
I was back and forth on it.  I landed on including `*args` to keep similar functionality found in `filter()`:
```python
nb.dcim.devices.all("ams01-leaf-08")
```

I seriously doubted anyone was doing `nb.dcim.devices.all("1.2", 100, 0)` with the current version.  There weren't any test harmed when making this change :)

I am not against it, but I do like the simplicity of just passing `*args` and `**kwargs` to `filter()`.